src/Todo.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import './Todo.css';
+
+interface TodoItem {
+  todo: string;
+  completed: boolean;
+}
+
+const Todo = () => {
+  const [todo, setTodo] = useState<string>('');
+  const [array, setArray] = useState<TodoItem[]>([]);
+
+  const addTodo = () => {
+    if (todo.trim() !== '') {
+      setArray([...array, { todo, completed: false }]);
+      setTodo('');
+    } else {
+      alert('Enter task');
+    }
+  };
+  const handleInput = (str: string) => {
+    setTodo(str);
+  };
+
+  const deleteTodo = (index: number) => {
+    const newArray = [...array];
+    newArray.splice(index, 1);
+    setArray(newArray);
+  };
+
+  const Completion = (index: number) => {
+    const newArray = [...array];
+    newArray[index].completed = !newArray[index].completed;
+    setArray(newArray);
+  };
+
+  return (
+    <div className="todo-container">
+      <h1>Todo app</h1>
+      <input type="text" value={todo} onChange={(e) => handleInput(e.target.value)} />
+      <button onClick={addTodo}> Add Todo</button>
+      <div>
+      <ul>
+        {array.map((item: TodoItem, index: number) => (
+          <li key={index}>
```

Comment:
Learning number 2 - *NEVER USE INDEX as key*

Now its your homework to find out why we should not use index as key. I will wait for your answer

Reply:
when we use index and add new element at beginning it will change the index of all the elements and re-render same for deletion part...and also it will create conflict with other sub array as index can be same